kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtComponentServiceImpl.java

```diff
@@ -732,18 +734,18 @@ public boolean updateProperties(GwtXSRFToken xsrfToken, String pid, Map<String,
 
             referenceInterfaces.forEach(reference -> {
```

Comment:
The code that creates the `referenceInterfaces` should only consider the references included in the definition of the component with `kura.service.pid=pid`. The `getPidsFromTarget()` method seem to ignore the `pid` argument completely.
This might cause incorrect interfaces to be added to the set and included later in the search.